runelite-client/src/main/java/net/runelite/client/plugins/config/ProfilePanel.java

```diff
@@ -660,6 +752,77 @@ private void switchToProfile(long id)
 		executor.submit(() -> configManager.switchProfile(profile));
 	}
 
+	private void unsetRsProfileDefaultProfile()
+	{
+		setRsProfileDefaultProfile(-1);
+	}
+
+	private void setRsProfileDefaultProfile(long id)
+	{
+		executor.submit(() ->
+		{
+			boolean switchProfile = false;
+			try (ProfileManager.Lock lock = profileManager.lock())
+			{
+				final String rsProfileKey = configManager.getRSProfileKey();
+				if (rsProfileKey == null)
+				{
+					return;
+				}
+
+				boolean changed = false;
+				for (final ConfigProfile profile : lock.getProfiles())
+				{
+					final List<String> defaultForRsProfiles = profile.getDefaultForRsProfiles();
+					if (defaultForRsProfiles == null)
+					{
+						continue;
+					}
+					changed |= profile.getDefaultForRsProfiles().remove(rsProfileKey);
+				}
+
+				if (id == -1)
+				{
+					log.debug("Unsetting default profile for rsProfile {}", rsProfileKey);
+				}
+				else
+				{
+					final ConfigProfile profile = lock.findProfile(id);
+					if (profile == null)
+					{
+						log.warn("setting nonexistent profile {} as default for rsprofile", id);
+						// maybe profile was removed by another client, reload the panel
+						reload(lock.getProfiles());
+						return;
+					}
+
+					log.debug("Setting profile {} as default for rsProfile {}", profile.getName(), rsProfileKey);
+
+					if (profile.getDefaultForRsProfiles() == null)
+					{
+						profile.setDefaultForRsProfiles(new ArrayList<>());
+					}
+					profile.getDefaultForRsProfiles().add(rsProfileKey);
+					switchProfile = true;
+					changed = true;
+				}
+
+				if (changed)
+				{
+					lock.dirty();
```

Comment:
it is safe to call .dirty() multiple times if you want, so you can just call it each place that it dirties the profiles.